File: leetcode/cpp/3750_Minimum_Number_of_Flips_to_Reverse_Binary_String.py
class Solution:
    def minimumFlips(self, n: int) -> int:
        bn = [(n >> i) & 1 for i in range(32)] 
        
        l = n.bit_length() - 1
        r = 0

        flips = 0

        while l > r:
            if bn[l] != bn[r]:
                flips += 2
            l -= 1
            r += 1

        return flips

File: leetcode/cpp/3750_Minimum_Number_of_Flips_to_Reverse_Binary_String.cpp
#include <bitset>
using namespace std;
class Solution {
public:
    int minimumFlips(int n) {
		bitset<32> bn(n);

		int l = 31 - __builtin_clz(n);
		int r = 0;

		int flips = 0;

		while (l > r) {
			if (bn[l] != bn[r])
				flips += 2;
			l--; r++;
		}
		return flips;
    }
};
